util/conditions/patch.go

```diff
@@ -49,7 +49,43 @@ const (
 	RemoveConditionPatch PatchOperationType = "Remove"
 )
 
+// NewPatchOperations returns the list of PatchOperations required to align source conditions to after conditions.
+func NewPatchOperations(before Getter, after Getter) (Patch, error) {
+	var patch Patch
+
+	if before == nil || reflect.ValueOf(before).IsNil() || after == nil || reflect.ValueOf(after).IsNil() {
```

Comment:
Out of curiosity when can `before != nil` and `reflect.ValueOf(before).IsNil() == true` happen?

Reply:
See the explanation in https://github.com/kubernetes-sigs/cluster-api/blob/3bff7da15910edf0b581875619d89d3bfbf59bd3/util/patch/patch.go#L302-L309 on why the reflection of the Value on the interface is needed when checking for `nil`.

That said, we can opt to let the function panic altogether, although that should be considered a breaking behavioral change

Reply:
Okay makes sense. But this essentially means to be safe we have to check all interface parameters of exported funcs additional with reflection?

Reply:
Ah! Very interesting. Never knew this before.

Reply:
We can, or decide to keep the nil exception. These panics should be caught during tests (unit, integration, e2e); unless the code isn't tested

Reply:
I personally prefer keeping the code as is which is way more readable over trying to cover this case.